packages/tx/src/EIP2930Transaction.ts

```diff
@@ -0,0 +1,385 @@
+    }
+
+    const values = rlp.decode(serialized.slice(1))
+
+    if (!Array.isArray(values)) {
+      throw new Error('Invalid serialized tx input. Must be array')
+    }
+
+    return EIP2930Transaction.fromValuesArray(values, opts)
+  }
+
+  // Create a transaction from a values array.
+  // The format is: chainId, nonce, gasPrice, gasLimit, to, value, data, access_list, [yParity, senderR, senderS]
+  public static fromValuesArray(values: (Buffer | AccessListBuffer)[], opts: TxOptions = {}) {
+    if (values.length == 8 || values.length == 11) {
+      const [chainId, nonce, gasPrice, gasLimit, to, value, data, accessList, v, r, s] = <
+        EIP2930ValuesArray
+      >values
+      const emptyBuffer = Buffer.from([])
+
+      return new EIP2930Transaction(
+        {
+          chainId: new BN(chainId),
+          nonce: new BN(nonce),
+          gasPrice: new BN(gasPrice),
+          gasLimit: new BN(gasLimit),
+          to: to && to.length > 0 ? new Address(to) : undefined,
+          value: new BN(value),
+          data: data ?? emptyBuffer,
+          accessList: accessList ?? emptyBuffer,
+          v: v !== undefined && !v.equals(emptyBuffer) ? new BN(v) : undefined,
+          r: r !== undefined && !r.equals(emptyBuffer) ? new BN(r) : undefined,
+          s: s !== undefined && !s.equals(emptyBuffer) ? new BN(s) : undefined,
+        },
+        opts ?? {}
+      )
+    } else {
+      throw new Error(
+        'Invalid EIP-2930 transaction. Only expecting 8 values (for unsigned tx) or 11 values (for signed tx).'
+      )
+    }
+  }
+
+  private constructor(txData: TxData, opts: TxOptions = {}) {
+    const { chainId, nonce, gasPrice, gasLimit, to, value, data, accessList, v, r, s } = txData
+
+    super({ nonce, gasPrice, gasLimit, to, value, data }, opts)
+
+    if (!this.common.eips().includes(2718)) {
+      throw new Error('EIP-2718 not enabled on Common')
+    } else if (!this.common.eips().includes(2930)) {
+      throw new Error('EIP-2930 not enabled on Common')
+    }
+
+    // TODO: verify the signature.
+
+    // check  the type of AccessList. If it's a JSON-type, we have to convert it to a buffer.
+
+    let usedAccessList
+    if (accessList && isAccessList(accessList)) {
+      this.AccessListJSON = accessList
+
+      const newAccessList: AccessListBuffer = []
+
+      for (let i = 0; i < accessList.length; i++) {
+        const item: AccessListItem = accessList[i]
+        //const addItem: AccessListBufferItem = []
+        const addressBuffer = toBuffer(item.address)
+        const storageItems: Buffer[] = []
+        for (let index = 0; index < item.storageKeys.length; index++) {
+          storageItems.push(toBuffer(item.storageKeys[index]))
+        }
+        newAccessList.push([addressBuffer, storageItems])
+      }
+      usedAccessList = newAccessList
+    } else {
+      usedAccessList = accessList ?? []
+      // build the JSON
+      const json: AccessList = []
```

Comment:
This stuff will be removed from the constructor

Reply:
This comment is probably outdated, I think it is fine in the constructor, it doesn't make much sense I think to create a private method here to handle this access list conversion.

Reply:
it might be nice to move this logic to a private method, just to keep the constructor light and readable